packages/features/bookings/Booker/components/OverlayCalendar/OverlayCalendarContainer.tsx

```diff
@@ -0,0 +1,153 @@
+import { useSession } from "next-auth/react";
+import { useRouter, useSearchParams, usePathname } from "next/navigation";
+import { useState, useCallback, useEffect } from "react";
+
+import dayjs from "@calcom/dayjs";
+import { useTimePreferences } from "@calcom/features/bookings/lib";
+import { classNames } from "@calcom/lib";
+import { useLocale } from "@calcom/lib/hooks/useLocale";
+import { trpc } from "@calcom/trpc/react";
+import { Button, Switch } from "@calcom/ui";
+import { Settings } from "@calcom/ui/components/icon";
+
+import { useBookerStore } from "../../store";
+import { OverlayCalendarContinueModal } from "../OverlayCalendar/OverlayCalendarContinueModal";
+import { OverlayCalendarSettingsModal } from "../OverlayCalendar/OverlayCalendarSettingsModal";
+import { useLocalSet } from "../hooks/useLocalSet";
+import { useOverlayCalendarStore } from "./store";
+
+export function OverlayCalendarContainer() {
+  const { t } = useLocale();
+  const [continueWithProvider, setContinueWithProvider] = useState(false);
+  const [calendarSettingsOverlay, setCalendarSettingsOverlay] = useState(false);
+  const { data: session } = useSession();
+  const setOverlayBusyDates = useOverlayCalendarStore((state) => state.setOverlayBusyDates);
+
+  const layout = useBookerStore((state) => state.layout);
+  const selectedDate = useBookerStore((state) => state.selectedDate);
+  const router = useRouter();
+  const pathname = usePathname();
+  const searchParams = useSearchParams();
+  const { timezone } = useTimePreferences();
+
+  // Move this to a hook
+  const { set, clearSet } = useLocalSet<{
+    credentialId: number;
+    externalId: string;
+  }>("toggledConnectedCalendars", []);
+  const overlayCalendarQueryParam = searchParams.get("overlayCalendar");
+
+  const { data: overlayBusyDates } = trpc.viewer.availability.calendarOverlay.useQuery(
+    {
+      loggedInUsersTz: timezone || "Europe/London",
+      dateFrom: selectedDate,
+      dateTo: selectedDate,
+      calendarsToLoad: Array.from(set).map((item) => ({
+        credentialId: item.credentialId,
+        externalId: item.externalId,
+      })),
+    },
+    {
+      enabled: !!session && set.size > 0 && overlayCalendarQueryParam === "true",
+      onError: () => {
+        clearSet();
+      },
+    }
+  );
+
+  useEffect(() => {
+    if (overlayBusyDates) {
+      const nowDate = dayjs();
+      const usersTimezoneDate = nowDate.tz(timezone);
+
+      const offset = (nowDate.utcOffset() - usersTimezoneDate.utcOffset()) / 60;
+      const offsettedArray = overlayBusyDates.map((item) => {
+        return {
+          ...item,
+          start: dayjs(item.start).add(offset, "hours").toDate(),
+          end: dayjs(item.end).add(offset, "hours").toDate(),
+        };
+      });
+      setOverlayBusyDates(offsettedArray);
```

Comment:
@CarinaWolli now using the timezone offset to move the events accordingly 

https://github.com/calcom/cal.com/pull/11693#discussion_r1350139381